R/plot_biomass.R

```diff
@@ -17,15 +17,25 @@
 #' @export
 #'
 plot_total_biomass <- function(dat,
-                                  show_warnings = FALSE,
-                                  units = NULL,
-                                  scaled = FALSE,
-                                  scale_amount = 1000,
-                                  ref_line = c("target", "MSY", "msy", "unfished"),
-                                  end_year = NULL,
-                                  relative = FALSE
+                               show_warnings = FALSE,
+                               units = NULL,
+                               scaled = FALSE,
+                               scale_amount = 1000,
+                               ref_line = c("target", "MSY", "msy", "unfished"),
+                               end_year = NULL,
+                               relative = FALSE,
+                               make_rda = FALSE,
+                               rda_folder = getwd()
 ){
 
+  # create plot-specific variables to use throughout fxn for naming and IDing
+  topic_label <- "biomass"
+  fig_or_table <- "figure"
+
+  # extract this plot's caption and alt text
+  caps_alttext <- extract_caps_alttext(topic_label = topic_label,
```

Comment:
Just a suggestion, but maybe instead of explicitly stating whether its a fig or table, you could add into the `extract_caps_alttext` fxn a way to recognize the object type and treat it as such from there similarly to `add_theme.R`

Reply:
What do you think about this: change extract_caps_alttext so that it reads the parent function name (e.g., plot_biomass), and if that parent function name contains "plot", it will signal that the parent function creates a figure? Could we commit to a naming system where the figure-making fxn names contain "plot"?

Link to how this could work: https://stackoverflow.com/questions/61469332/determine-function-name-within-that-function